codegen/builtin_fs.go

```diff
@@ -494,6 +517,13 @@ func (l Label) Call(ctx context.Context, cln *client.Client, ret Register, opts
 	}
 
 	fs.Image.Config.Labels[key] = value
+
+	numHistory := len(fs.Image.History)
+	if numHistory > 0 && strings.HasPrefix(fs.Image.History[numHistory-1].CreatedBy, "LABEL") {
+		fs.Image.History[numHistory-1].CreatedBy = fmt.Sprintf("%s %s=%s", fs.Image.History[numHistory-1].CreatedBy, key, value)
```

Comment:
Can you explain why we replace rather than append the label in this case? It might be worth a comment in the code.

Reply:
In Dockerfile it is `LABEL key=value key2=value2 ...` whereas in HLB it is `fs label(string key, string value)` singular. So we need to merge to produce the same history.

I also tested `LABEL samekey=foo samekey=bar` with Dockerfile and it does show up as is in the history without deduplication.

Will add a comment.

Reply:
Ah, I missed that we were including the existing contents. How about this to make it slightly more obvious and succinct:

```suggestion
		fs.Image.History[numHistory-1].CreatedBy += fmt.Sprintf(" %s=%s", key, value)
```